[Chen Rui Han] iP

rui-han-crh opened the pull request:
### Duke
> "There's a door to center stage and we have the key" – Sana ([source](https://youtu.be/jHibTDJlTJw?t=113))

Duke... Uh... is a checklist.
- Is a checklist
- Can have multiple checklists, each checklist has it's own tasks
- It's ~~cool~~ _very cool_

It supports 3 types of tasks and this is what a check list looks like
1. [T][ ] read book
2. [D][ ] return book (by: 13 Sep 2022 02:55 PM)
3. [E][ ] book fair (at: 14 Sep 2022 09:30 AM)

OMG I LOVE **EMOJIS** :laughing::laughing::laughing:

Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)


If you Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

rui-han-crh committed:
Completes level-1 increment

rui-han-crh committed:
Finishes Level-2 Increment

rui-han-crh committed:
Finishes Level-3

rui-han-crh committed:
Completes Level-4

rui-han-crh committed:
Finishes Level-5

rui-han-crh committed:
Actually finishes Level-5, the last one finishes automated test ui testing

rui-han-crh committed:
Finishes Level-6, adds delete functionality

rui-han-crh committed:
Reorganises functions
Adds tests
Retargets testing target

rui-han-crh committed:
Level 7, adds serialisation and deserialisation abilities

rui-han-crh committed:
applies date and time parsing, sorting and filtering

rui-han-crh committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/DukeProgram/Checklist.java

rui-han-crh committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/DukeProgram/Checklist.java

rui-han-crh committed:
Resolves function names

rui-han-crh committed:
Merge branch 'master' of https://github.com/rui-han-crh/ip

# Conflicts:
#	src/main/java/DukeProgram/Checklist.java

rui-han-crh committed:
Merge branch 'add-gradle-support' of https://github.com/nus-cs2103-AY2223S1/ip

rui-han-crh committed:
Removes unused imports

rui-han-crh committed:
Adds Gradle Support

rui-han-crh committed:
Extracts functions into classes and packages them

rui-han-crh committed:
Reorganises invocation flow of command classes

The last commit had commands with functions that do not serve concrete purposes. This commit ensures that execute() is used to perform looping instructions until termination and parse(String) is used to parse new commands that will be interpreted within this class, either from the execute method, or from an external user that calls parse on this object.

rui-han-crh committed:
Updates .gitignore

The previous git ignore does not ignore the MANIFEST.mf files. This commit makes the git ignore ignore this specific file by name

rui-han-crh committed:
Adds JavaDocs to most methods

The previous iteration did not have any documentation
Therefore, this iteration adds documentations to nearly every method in every class.

rui-han-crh committed:
Refactors package names to lowercase

The previous iteration had all package names in uppercase. This commit fixes all packages to lowercase.

rui-han-crh committed:
Adds Find Functionality

The previous commit does not have any ability to query task names
This commit adds the ability to find any consecutive characters and return the entire task if it exists in its full name

rui-han-crh committed:
Adds Find Functionality

The previous commit does not have any ability to query task names
This commit adds the ability to find any consecutive characters and return the entire task if it exists in its full name

rui-han-crh committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/DukeProgram/Commands/Task/AddTaskCommand.java
#	src/main/java/DukeProgram/Commands/Task/DeleteTaskCommand.java
#	src/main/java/dukeprogram/facilities/TaskList.java
#	src/main/java/dukeprogram/parser/Parser.java
#	src/main/java/dukeprogram/storage/SaveManager.java
#	src/main/java/dukeprogram/ui/UserInterface.java

rui-han-crh committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/DukeProgram/Commands/Task/TaskPageCommand.java
#	src/main/java/dukeprogram/facilities/TaskList.java

rui-han-crh committed:
Refactors Packages Names

The last merge overwrote some package names back to capital letters.
This commit reverses the overwrite by renaming offending packages back to lowercase.

rui-han-crh committed:
Refactors the structure of Commands

The last merge broke several restructured commands. This commit adds them back.

rui-han-crh committed:
Refactors the structure of Commands

The last merge broke several restructured commands. This commit adds them back.

rui-han-crh committed:
Resolves Compile Issues

Package naming was not synchronised properly across the Intellij package names and the actual package names. There were also several overwritten errors due to merge that caused progress loss.

This commit revises everything and compiles the program as it should have before.

rui-han-crh committed:
Uses Checkstyle

rui-han-crh committed:
Revert "Uses Checkstyle"

This reverts commit 56d46e14ae8fae98224ff3b227235deaa118253e.

rui-han-crh committed:
Fulfils Level 10

The last iteration had no visual GUI
This is a massive change by adding a GUI, recognising the user by computer username and saving to task list automatically.

rui-han-crh committed:
Merge branch 'master' into branch-Level-10

The previous iteration is severely outdates.
The incoming branch has all the necessary changes.
We will be dumping all new changes into master by overwriting all files.

rui-han-crh committed:
Updates UI

This commits makes the UI look nicer with more colors

rui-han-crh committed:
Updates UI

Changes fonts and format of dialog boxes

lilythchu commented on src/main/java/dukeprogram/command/ListTasksCommand.java:
```diff
@@ -0,0 +1,39 @@
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+public class ListTasksCommand extends Command {
```
Maybe you could  Javadoc comment for this class

lilythchu commented on src/main/java/dukeprogram/parser/Parser.java:
```diff
@@ -0,0 +1,20 @@
+package dukeprogram.parser;
+
+import dukeprogram.command.Command;
```
This class is also unused

lilythchu commented on src/main/java/dukeprogram/DialogBox.java:
```diff
@@ -0,0 +1,101 @@
+import javafx.scene.layout.BackgroundFill;
+import javafx.scene.layout.CornerRadii;
+import javafx.scene.layout.HBox;
+import javafx.scene.layout.Region;
```
It seems like this class is unused so maybe you could remove this line.

lilythchu commented on src/main/java/dukeprogram/InternalAction.java:
```diff
@@ -0,0 +1,57 @@
+package dukeprogram;
+
+import javax.print.attribute.standard.RequestingUserName;
```
This class is also unused.

lilythchu commented on src/main/java/dukeprogram/command/ListTasksCommand.java:
```diff
@@ -0,0 +1,39 @@
+
+import dukeprogram.InternalAction;
+import dukeprogram.facilities.TaskList;
+import dukeprogram.userinterface.Ui;
```
This is also unused.

lilythchu reviewed (commented):
Your code looks good! Just have some unused imports.

rui-han-crh committed:
Updates UI

Creates a more aesthetically pleasing UI design. Involves visual changes only with no functionality changes.

rui-han-crh committed:
Updates the Gradle Build File

The last commit had no build version specified. This commit specifies a build version.
This commit also changes the directory name of the Save File

rui-han-crh committed:
Modifies JavaDocs

The last version did not have appropriate JavaDocs for ListTasksCommand class header. This commit adds the comments.
In additional, there were unused imports that were not removed - this commit will remove them.

rui-han-crh committed:
Adds Assertions at Critical Points

To ensure the accuracy of this codebase and minimise error occurances, let us add asserts to specific points in the code. They signify the tautological assumptions at each point of execution, failing which implies that there is are unaccounted corner cases, which should not have led to that execution point.
Doing this will help us reduce the amount of time spent on future debugs by making our assumptions clear.

rui-han-crh committed:
Applies the use of Streams

The last commit already had streams used. This commit applies the use of streams more frequently, particularly whenever a for or foreach loop is encountered. This helps in the code being read more naturally.

rui-han-crh committed:
Merge pull request #1 from rui-han-crh/A-Streams

Applies the use of Streams

rui-han-crh committed:
Merge branch 'master' into A-CodeQuality

rui-han-crh committed:
Merge branch 'master' into A-Assertions

rui-han-crh committed:
Merge pull request #2 from rui-han-crh/A-CodeQuality

Verify the Adherance to Code Quality

rui-han-crh committed:
Merge branch 'master' into A-Assertions

rui-han-crh committed:
Merge pull request #3 from rui-han-crh/A-Assertions

Adds Assertions at Critical Points

rui-han-crh committed:
Adds YAML File for CI/CD

rui-han-crh committed:
Fixes Adherance to Code Quality

rui-han-crh committed:
Add New Features

Updates the task list ui

rui-han-crh committed:
Refactors Code for Single Input Parsing

Before this commit, the code was set up in a way that requires multi staged inputs, such as first entering into a task stage, then executing commands in the task stage. As per the suggestions in the guidelines, this is now changed to single line input and will now require execution in one line only.

For example: task add todo do laundry

Some exceptions are reserved for confirmation commands, but need not be answered and can be ignore.

For example: task delete all
> Are you sure you want to delete all tasks?
One can choose to say yes, no, or ignore this prompt altogether and issue another command, on which the previous instruction will be ignored and the new command will be executed instead.

Updates to the GUI added: Dark Mode

rui-han-crh committed:
Applies Better Transitions

This commit will apply nicer looking transitions to the GUI such as horizontal and vertical sliding, window resizing and anchoring changes.
A new background is also added.
Widget version of the task list was removed due to incompatibility.

rui-han-crh committed:
Repackages classes and fixes imports

For better adherance to checkstyle, as well as a tidier package hierachy, the codebase was refactors to put Task related classes into a tasks package and UI related classes into userinterface package.

rui-han-crh committed:
Update README.md

rui-han-crh committed:
Fixes GUI and Adds More Functionalities

GUI smooth scrolling and sliding has been fixed. The window will now resize chat bubbles based on their text length and font.
The Loan recording function has been added.
Added the ability to change your name.

rui-han-crh committed:
Adds Readme and UI to Docs Folder

rui-han-crh committed:
Updates README.md

rui-han-crh committed:
Update README.md

rui-han-crh committed:
Update README.md

rui-han-crh committed:
Revert "Update README.md"

This reverts commit 34415a4315302a7a0ae31eaaa60601db66c5c571.

rui-han-crh committed:
Merge branch 'master' of https://github.com/rui-han-crh/ip

rui-han-crh committed:
Update README.md

rui-han-crh committed:
Update README.md

rui-han-crh committed:
Update README.md

rui-han-crh committed:
Modifies Help Command to Redirect to User Guide

rui-han-crh committed:
Updates Widget GUI

rui-han-crh committed:
Adds Ability to Open Folder for Setting Pictures

rui-han-crh committed:
Update documentation

rui-han-crh committed:
Update Documentation

rui-han-crh committed:
Updates Documentaiton

rui-han-crh committed:
Update Documentation

rui-han-crh committed:
Update README.md

rui-han-crh committed:
Update Documentation

rui-han-crh committed:
Updates Application Icon and Name

rui-han-crh committed:
Update README.md

rui-han-crh committed:
Fix an Issue with Linux systems Not Loading Background

The last version had the background in a jpg format, which for some reason is not able to be loaded properly on Linux based system. This version converts the image to PNG format.

rui-han-crh committed:
Update README.md

Fixes typos.

rui-han-crh committed:
Upgrades JavaFX to 11

Encountering issue with Linux system not being able to be run with Djdk.gtk.version=2. Hopefully, this fixes it. I'll have to see when I deploy to Linux Ubuntu.

rui-han-crh committed:
Revert Last Change in build.gradle

rui-han-crh committed:
Fixes Critical Typo Issue with SaveFile Name

Changed saveFile to savefile on line 31 of LoadUserCommand. Note the small letter f in the middle of the word

rui-han-crh committed:
Adds More Test Cases

Adds more test cases pertaining to tasklist, loans and date time parsing

rui-han-crh committed:
Fixes Checkstyle Issues

rui-han-crh committed:
Fixes Checkstyle Issues

rui-han-crh committed:
Fixes Checkstyle Issues related to the testing Package

rui-han-crh committed:
Fixes Checkstyle Issues Related to DateTimeParsingTests

rui-han-crh committed:
Changes UI Appearance, Exits System Completely

Changes the color of the task label widget
Disables the ability select the checkbox of tasks in the GUI

Applies exit with system code 0 on bye command

rui-han-crh committed:
Update README.md
